src/test/java/uk/gov/ons/ctp/integration/contactcentresvc/service/impl/CaseServiceImplGetUACTest.java

```diff
@@ -0,0 +1,174 @@
+    Mockito.doThrow(new ResponseStatusException(HttpStatus.NOT_FOUND))
+        .when(caseServiceClient)
+        .getCaseById(UUID_0, false);
+    Mockito.when(dataRepo.readCachedCaseById(UUID_0)).thenReturn(Optional.of(new CachedCase()));
+    target.getUACForCaseId(UUID_0, new UACRequestDTO());
+  }
+
+  @Test(expected = ResponseStatusException.class)
+  public void testGetUAC_caseServiceNotFoundException_noCachedCase() throws Exception {
+    Mockito.doThrow(new ResponseStatusException(HttpStatus.NOT_FOUND))
+        .when(caseServiceClient)
+        .getCaseById(UUID_0, false);
+    Mockito.when(dataRepo.readCachedCaseById(UUID_0)).thenReturn(Optional.empty());
+    target.getUACForCaseId(UUID_0, new UACRequestDTO());
+  }
+
+  @Test(expected = ResponseStatusException.class)
+  public void testGetUAC_caseServiceResponseStatusException() throws Exception {
+    Mockito.doThrow(new ResponseStatusException(HttpStatus.I_AM_A_TEAPOT))
+        .when(caseServiceClient)
+        .getCaseById(UUID_0, false);
+    target.getUACForCaseId(UUID_0, new UACRequestDTO());
+  }
+
+  @Test
+  public void shouldRejectCeManagerFormFromUnitRegionE() {
+    CaseContainerDTO dto = mockGetCaseById("CE", "U", "E");
+    assertThatCeManagerFormFromUnitRegionIsRejected(dto);
+  }
+
+  @Test
+  public void shouldRejectCeManagerFormFromUnitRegionW() {
+    CaseContainerDTO dto = mockGetCaseById("CE", "U", "W");
+    assertThatCeManagerFormFromUnitRegionIsRejected(dto);
+  }
+
+  @Test
+  public void shouldRejectCeManagerFormFromUnitRegionN() {
+    CaseContainerDTO dto = mockGetCaseById("CE", "U", "N");
+    assertThatCeManagerFormFromUnitRegionIsRejected(dto);
+  }
+
+  @Test
+  public void shouldRejectCeManagerFormFromEstabRegionN() {
+    CaseContainerDTO dto = mockGetCaseById("CE", "E", "N");
+    assertThatInvalidLaunchComboIsRejected(
+        dto,
+        "All Northern Ireland calls from CE Managers are to be escalated to the NI management team.");
+  }
+
+  @SneakyThrows
+  private void assertThatInvalidLaunchComboIsRejected(CaseContainerDTO dto, String expectedMsg) {
+    try {
+      doGetUACTest(false, dto, FormType.C);
+      fail();
+    } catch (CTPException e) {
+      assertEquals(Fault.BAD_REQUEST, e.getFault());
+      assertTrue(e.getMessage(), e.getMessage().contains(expectedMsg));
+    }
+  }
+
+  @SneakyThrows
+  private void assertThatCeManagerFormFromUnitRegionIsRejected(CaseContainerDTO dto) {
+    assertThatInvalidLaunchComboIsRejected(
+        dto, "A CE Manager form can only be launched against an establishment address not a UNIT.");
+  }
+
+  private void doGetUACTest(String caseType, boolean individual) throws Exception {
+    CaseContainerDTO caseFromCaseService = mockGetCaseById(caseType, "U", A_REGION.name());
+    doGetUACTest(individual, caseFromCaseService, FormType.H);
+  }
+
+  private void doGetUACTest(
+      boolean individual, CaseContainerDTO caseFromCaseService, FormType formType)
```

Comment:
caseFromCaseService isn't used

Reply:
Carelessness from using CaseServiceImplLaunchTest which tests some of the same business logic as a template. Checked the method body and forgot the method signature. Actually allows for a lot of further simplification. Changed.